### ISSUE TITLE
Recursive resource lookup bug - Infinite recursion during resource lookup within System.Private.CoreLib.

### ISSUE THREAD

**Issue by 304NotModified** [NLog/NLog.Web#210]
This is a bug in dotnet.

See https://github.com/NLog/NLog.Web/issues/206#issuecomment-337542395 and https://github.com/dotnet/coreclr/issues/12668

**Comment by 304NotModified** [comment]
Fix needed. PR more then welcome!

**Comment by 304NotModified** [comment]
see also dotnet/coreclr#14033

**Comment by 304NotModified** [comment]
work around:

```
Thread.CurrentThread.CurrentUICulture = new CultureInfo("en-US");
Thread.CurrentThread.CurrentCulture = new CultureInfo("en-US");
```

**Comment by liu-ning** [comment]
thx

**Comment by Scandal-UK** [comment]
I only get this problem when I deploy to CentOS (which is my production environment).

Unfortunately the workaround does not work for me as my "locale" in CentOS is en-GB, setting the culture to this does not work and I do not want to change the culture in my app to en-US!

Is there another possible workaround?

**Comment by 304NotModified** [comment]
I think I fixed this another way in beta 4 (https://www.nuget.org/packages/NLog.Web.AspNetCore/4.5.0-beta04) 

Could you please test that?

**Comment by swisszeni** [comment]
Beta 4 still has the same problem for me.

**Comment by yhnbgfd** [comment]
Beta 4 still has the same problem for me when i publish my project.
(NLog.Web.AspNetCore 4.5.0-beta04;NLog 4.5.0-beta07)

Adding "new ArgumentException();" to the start of the Main() method works.

**Comment by danmosemsft** [comment]
Fix released https://github.com/dotnet/corefx/releases/tag/v2.0.3

**Comment by smoogipoo** [comment]
Confirming that this is fixed for us. Thanks a lot @danmosemsft !

**Comment by danmosemsft** [comment]
You're welcome, I wish it had not taken so long, but we should have some of the kinks worked out of the servicing process now.

**Comment by agentschmitt** [comment]
Can also confirm that 2.0.3 fixes the issue. It's now working without the workaround.

**Comment by 304NotModified** [comment]
OK, thanks for the feedback! 

Closing this one.

**Comment by snakefoot** [comment]
For standalone apps it's necessary to set this attribute in the .csproj to reference a specific version:

```
<PropertyGroup>
  <RuntimeFrameworkVersion>2.0.5</RuntimeFrameworkVersion>
</PropertyGroup>
```

Or `<RuntimeFrameworkVersion>2.0.*</RuntimeFrameworkVersion>` to target the latest published 2.0.x runtime. See also https://github.com/dotnet/corefx/issues/26292
